autoload/jsdoc.vim

```diff
@@ -459,7 +459,11 @@ function! jsdoc#insert() abort
         if l:arg['type'] != ''
           let s:candidate_type = l:arg['type']
         endif
-        let l:argType = input('Argument "' . l:arg['val'] . '" type: ', '', 'custom,jsdoc#listDataTypes')
+        if s:candidate_type == ''
+            let l:argType = input('Argument "' . l:arg['val'] . '" type: ', '', 'custom,jsdoc#listDataTypes')
```

Comment:
Sorry for late review.
Alway skip `type` is not good.
If user want input type manually, how to do that?

Reply:
First things first. This is my first time doing a PR on a vim plugin so I may be wrong.

My understanding of my proposal is that the type isn't asked to the user __only__ if it's already provided in the code itself otherwise the plugin __should keep asking__ for the type.

I can't came up with a scenario where the user write a code defining a type then in the documentation he want to specify a different type. It doesn't make sense to me based on my experience.
I'm not pretending that my use case should be the only one but as far as I'm concerned it's the right one.

If my PR doesn't follow this flow then I've done a mistake otherwise it's the right modification to me and I'm using it since I've written it.

P.S. don't worry and take your time we're not on the clock ;)

Reply:
Thank you for comment.

At first I don't want break backword compatible.
Adding new global option is not a good idea but it much better than BC break.

>I can't came up with a scenario where the user write a code defining a type then in the documentation he want to specify a different type. It doesn't make sense to me based on my experience.

For instance(in my case),
```
function foo (arg1: string, arg2: { foo, bar }) {
}
```

```
  /**
   * foo
   *
   * @param {string} arg1
   * @param {Object} arg2
   * @returns {undefined}
   */
function foo (arg1, arg2: { foo, bar }) {
}
```
`arg2` would be `@param {foo, bar}` but user may want input just `Object`.
(I know current implementaion have a problem with object literal)
I want user can choose input or not by themself.

Or how about adding new global option something like `g:jsdoc_input_param_description = 1` and if `g:jsdoc_allow_input_prompt = 0` than prompt description.
If user don't want prompt, just set `g:jsdoc_input_parm_description = 0` and this is not break BC.

Reply:
I don't want to sound rude but English isn't my first language.
I'm not sure about your example because `@param {Object}`is a generic object while `{foo, bar}`is an object with two property (foo and bar) both of type `any` so in your documentation you are writing something wrong.

You should do something like
```
interface customObj {
    foo: any;
    bar: any;
}
  /**
   * foo
   *
   * @param {string} arg1
   * @param {customObj} arg2
   * @returns {undefined}
   */
function foo (arg1: string, arg2: customObj) {
}
```
but I don't want to force you to agree with me.

If you want to add a new global option then I'm good with it otherwise I could stick with a fork of your plugin.

Reply:
@Edo78 
Sorry for late late response

Please add global option `g:jsdoc_input_param_description` and default is `0`.
And than I'll include this.

>You should do something like

That's users choice. jsdoc.vim should not force it.

Reply:
Sorry for the late response.
It was a long time (I stop worked with typescript) but now I'm willing to complete this PR.

Can you recap the flow you'd like to include?

As far as I get you proposed a global option `g:jsdoc_input_param_description` with a default of `0` and a flow

> Or how about adding new global option something like g:jsdoc_input_param_description = 1 and if g:jsdoc_allow_input_prompt = 0 than prompt description.
If user don't want prompt, just set g:jsdoc_input_parm_description = 0 and this is not break BC.

If this is still ok for you I can complete it in the weekend or in the coming week.

Let me know and sorry again for being so late.